### ISSUE TITLE
Microsoft.Maui.Controls 8.0.90 (Latest stable) Ruins Entry Controls

### ISSUE THREAD

**Issue by Dannyk90** [dotnet/maui#24783]
### Description

If you update the Maui Controls NuGet to the latest stable (currently 8.0.90), all Entry controls become very narrow and nothing will fix it apart from downgrading, I just went back to 8.0.72 and it is fine again.

This appears to only affect Windows. Not IOS or Android.

### Steps to Reproduce

1. Create a new MAUI project
2. Add an Entry control to the home page xaml.
3. Launch and see that it looks fine
4. Upgrade the Maui Controls NuGet to the latest stable 8.0.90
5. Launch project again and see that the Entry looks awful now.

### Version with bug

8.0.90

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.72

### Affected platforms

Windows

### Affected platform versions

Windows

### Did you find any workaround?

Downgrading the NuGet

**Comment by bronteq** [comment]
@MartyIX 8.0.90 is not yet available in the issue template

**Comment by baaaaif** [comment]
Can confirm this on our App. SR9 cannot be used at the moment.
Editor does not have this bug.

![image](https://github.com/user-attachments/assets/d9770ae5-f5dc-4256-bf77-a554f311ec18)

**Comment by PureWeen** [comment]
> Interestingly, I don't observe this behavior in my app.

Yea, @Foda and I both tested this as well and couldn't reproduce

I realize the repro steps seem simple, but if you could still attach a repro just so we can validate that would be helpful.

**Comment by Dannyk90** [comment]
@PureWeen , I can no longer recreate the problem. I have since updated visual studio and the .net sdk (to fix another issue).
@baaaaif maybe you can try updating those too and see if that resolves it for you too

**Comment by Dannyk90** [comment]
@baaaaif , the problem does happen again if I install Telerik NuGet **and** put .UseTelerik() in MauiProgram.cs

**Comment by baaaaif** [comment]
@TelerikMauiTeam sorry for pinging, but might be in your interest

**Comment by ninachen03** [comment]
This issue has been verified using Visual Studio 17.12.0 Preview 2.0(8.0.90 & 8.0.80 &8.0.72). Cannot repro this issue at windows platform.

**Comment by drasticactions** [comment]
> This issue has been verified using Visual Studio 17.12.0 Preview 2.0(8.0.90 & 8.0.80 &8.0.72). Cannot repro this issue at windows platform. By the way,I can't use the Telerik.UI.for.Maui,Is there anything special about it? ![image](https://private-user-images.githubusercontent.com/128687574/368827557-242e575a-18cf-4a0f-a17a-6af8a4bc1a6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3NjUwMjIsIm5iZiI6MTcyNjc2NDcyMiwicGF0aCI6Ii8xMjg2ODc1NzQvMzY4ODI3NTU3LTI0MmU1NzVhLTE4Y2YtNGEwZi1hMTdhLTZhZjhhNGJjMWE2Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxOVQxNjUyMDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NGE0NWViZGI2MDlkOGZlYWNmZTA2MmRiNDg3OWRkM2UzNjg4ZDEwMzFmMjcxNzRlMDVjNWM0YTg5ZTIyYjljJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ml8Aj00V29EcEyzVNNKl5-2pTQq96rutTEvLYaF0gEw)

Yes, they are not on Nuget, I believe you need to download them from [Telerik's website](https://www.telerik.com/maui-ui)

**Comment by Dannyk90** [comment]
Telerik have their own NuGet source

**Comment by cmpalmer66** [comment]
I verified that, starting from a fresh templated project and updating Maui to 8.0.90, the problem did not appear until I added a reference to Telerik's UI for .NET Maui 7.1.0.

Before:
![image](https://github.com/user-attachments/assets/0e5e6093-7a0e-46c0-bc9c-501b9d79a751)

After adding the Telerik Nuget package (but not adding UseTelerik() to the Builder):
![image](https://github.com/user-attachments/assets/ecb5b489-15dd-4b26-a1cd-9dd31f6476f5)

I've attached my sample project with the Telerik reference. You can run it to verify that the problem only appears on WinUI and goes away when the Telerik reference is removed.
[WinUIEntryBug.zip](https://github.com/user-attachments/files/17079081/WinUIEntryBug.zip)

**Comment by PureWeen** [comment]
I'm going to close this issue for now unless I hear otherwise from Telerik

Can you create an issue using their support system and reference this issue?

**Comment by cmpalmer66** [comment]
I did create an issue for Telerik. I am curious as to the cause, though, because the problem arose on the release of Maui 8.0.90, not on the release of the Telerik 7.1.0, which has been out and working for a while.

https://feedback.telerik.com/maui/1665106-version-7-1-0-and-maui-version-8-0-90-break-sizing-of-winui-entry-controls

**Comment by malsabi** [comment]
When I updated to MAUI 8.0.90 I had the same Issue and I was not using Telerik. Maybe this issue happens when you attempt to make the Entry borderless? cause my entry is borderless by setting BorderThickness to 0 for Windows platform. I had to downgrade back to 8.0.80

**Comment by LanceMcCarthy** [comment]
HI Shane, it is a holiday in Sofia today, but I'll make sure the team take a look at it tomorrow morning (about 1 am Redmond time tonight) follows up in that public item once there is a determination of what stye thread 

@malsabi This is a very important insight that could swing the investigation in the right direction. When adding the Telerik package, you are applying some standard implicit styles, and there are 0-width BordherThickness values in the styles. Can you please share the XAML (and any associated Style) for the Entry that is affected?

**Comment by LanceMcCarthy** [comment]
Hi @PureWeen, I can reproduce the issue without Telerik involvement. As @malsabi suspected, this is indeed related to when the `StrokeThickness` property is set to `0`.

The reason it initially seemed to be related to Telerik is because we have implicit styles to standardize the styling. when you add our package, the Entry's default style has StrokeThickness set to 0 by default, while the Microsoft File > New > Maui project has this value set to 1 by default.

Here's all you need validate the same result:

```xaml
<?xml version="1.0" encoding="utf-8"?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="WinUIEntryBug.MainPage">

    <ContentPage.Resources>
        <!-- Moved from Styles/Styles.xaml-->
        <Style TargetType="Border">
            <Setter Property="Stroke" Value="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource Gray500}}" />
            <Setter Property="StrokeShape" Value="Rectangle"/>

            <!-- Reproduced -->
            <Setter Property="StrokeThickness" Value="0"/>

            <!-- Default values used in File > New Project 
            <Setter Property="StrokeThickness" Value="1"/> -->
        </Style>
    </ContentPage.Resources>
    <ScrollView>
        <VerticalStackLayout
            Padding="30,0"
            Spacing="25">

            <Border>
                <Entry />
            </Border>

            <Grid>
                <Grid.ColumnDefinitions>
                    <ColumnDefinition Width="*" />
                </Grid.ColumnDefinitions>

                <Entry Grid.Column="0"></Entry>
            </Grid>

            <Frame>
                <Entry />
            </Frame>

            <Entry />
        </VerticalStackLayout>
    </ScrollView>

</ContentPage>
```

I am still waiting on the engineering team to verify my findings to close the Telerik Feedback Portal item, I think this is enough information for this item to be re-opened and moved forward.

**Comment by cmpalmer66** [comment]
I _may_ be seeing a similar (related?) issue with the widths of the CheckBox controls in WinUI. They are stuck at about a couple of cm wide (with blank space to the right of the checkbox) and setting their WidthRequest or changing their parent container doesn't seem to affect how they render. I had to put the checkbox in grid cell with a ColumnDefinition with a width of "35" to clip the extra space off the Checkbox.

Later tonight, I'm going to try to isolate this problem. It may be related to this issue, or it may be a separate issue. The reason I suspected they might be related is that the extent of the CheckBox control is about the same width that the Entry control is locked to, and, like the Entry, it doesn't respect WidthRequests etc.

**Comment by malsabi** [comment]
@PureWeen Can you please reopen this issue and fix it as soon as possible

**Comment by LanceMcCarthy** [comment]
Hi folks, after digging into the Live Visual Tree Explorer, I have some more evidence on what is responsible.  Please see my 1 minute explanation with evidence at runtime [https://www.screencast.com/t/v7JTtug3U](https://www.screencast.com/t/v7JTtug3U)

First, @cmpalmer66 the reason using HorizontalOption=Stretch has no effect is because it is already still using Strech. see my video for an explanation.

Now, what is responsible for the inner area? It depends on what part of the control we need to look at next. I highly suspect Border to be responsible, but need more time to dig into the Handler (which is of type `MauiPasswordBox` on Windows)

**Comment by sbloomztf** [comment]
Same issue encountered today, and yes I am using telerik controls, but this is a new bug introduced in 8.0.90.  It was okay with 8.0.82.

**Comment by LanceMcCarthy** [comment]
@sbloomztf As a temporary workaround, you can use RadEntry instead of Entry, if you have to use 8.0.90. See Didi's response here https://feedback.telerik.com/maui/1665106-version-7-1-0-and-maui-version-8-0-90-break-sizing-of-winui-entry-controls 

You can follow that item to see what our findings will be, too.

**Comment by malsabi** [comment]
Why is this still considered close? Should I make a new issue? I can link a new repo without Telerik showing its still bugged when using Custom Entry with Border Thickness being set to 0. Please let me know @PureWeen

**Comment by drasticactions** [comment]
@LanceMcCarthy Could you please create an example repo of this happening with the standalone controls? I'm unsure of what exactly you're setting or wrapping the views around, but as you said you can get it to happen outside of Telerik then this should be open until shown otherwise.

**Comment by LanceMcCarthy** [comment]
@malsabi Can you please share the demo project that reproduces the issue with @drasticactions ?

I do not have something handy, and what you just described sounds like it's the underlying problem.

**Comment by malsabi** [comment]
@LanceMcCarthy  Sure here is the result without using Telerik.

I have added implicit styles in the Platforms/Windows/app.xaml for the TextBox.

Repo: https://github.com/malsabi/EntryIssue

Result:
![Screenshot_1](https://github.com/user-attachments/assets/b50c6e25-49bc-4bf7-8eba-64d2266a52dd)

**Comment by drasticactions** [comment]
Yeah, there's an issue here. But it's... weird. As far as I can tell, the act of setting TextBox.BorderThickness _period_ causes the Textbox to truncate. Setting it to 0, 1, 50, doesn't matter. It will always truncate. If you remove the style, it expands within the container.

<img width="596" alt="image" src="https://github.com/user-attachments/assets/abda1e03-5b53-498c-b683-2ecec30d5e28">

With the Live Visual Tree, you can inspect the inner WinUI Elements. The Textbox itself is full width, but the elements inside it are truncated. If you set a Width on the Border, it does expand to match the specific width and the other elements in it match. But only with Border. Any other element inside it won't force it to expand.

**Comment by malsabi** [comment]
@drasticactions Yes, I noticed the same thing. When setting BorderThickness to 0 in the EntryHandler.cs class, it doesn't have any effect unless a style is applied, which then causes the same issue. This behavior started with version 8.0.90.

Note: I would expect setting BorderThickness to 0 in the EntryHandler to behave the same way as setting it through a style. I'm not sure why the border thickness only changes when a style is added.

**Comment by drasticactions** [comment]
I think I figured it out

https://github.com/dotnet/maui/pull/24485

This PR regressed it. I _think_ it's forcing the delete button to truncate because it's setting the width to Auto in the inner column, and the setting of the Border width happens to trigger the inner logic within the control to resize it. 

If this code is reverted, all is right again and it's full width.

**Comment by LanceMcCarthy** [comment]
Nice find @drasticactions!

This also explains why it initially appeared to be a Telerik issue (because we do have native styles to adhere to cross platform consistency).
